tst/ProducerFunctionalityTest.cpp

```diff
@@ -591,6 +591,73 @@ TEST_F(ProducerFunctionalityTest, intermittent_producer_verify_eofr_sent)
     mStreams[0] = INVALID_STREAM_HANDLE_VALUE;
 }
 
+TEST_F(ProducerFunctionalityTest, intermittent_producer_verify_eofr_sent_multi_track)
+{
+    STREAM_HANDLE streamHandle = INVALID_STREAM_HANDLE_VALUE;
+    UINT32 i, errCount;
+    UINT32 totalFrames = 510;
+    UINT64 startTime, delay;
+    mFrame.duration = 100 * HUNDREDS_OF_NANOS_IN_A_MILLISECOND;
+    mKeyFrameInterval = 10;
+    UPLOAD_HANDLE streamUploadHandle = 0;
+
+
+    PStreamInfo pStreamInfo;
+
+    EXPECT_EQ(STATUS_SUCCESS,
+            createRealtimeAudioVideoStreamInfoProvider(TEST_STREAM_NAME,
+                                                       TEST_RETENTION_PERIOD,
+                                                       TEST_STREAM_BUFFER_DURATION,
+                                                       &pStreamInfo));
+
+    EXPECT_EQ(FRAME_ORDERING_MODE_MULTI_TRACK_AV_COMPARE_PTS_ONE_MS_COMPENSATE_EOFR, pStreamInfo->streamCaps.frameOrderingMode);
+    EXPECT_EQ(2, pStreamInfo->streamCaps.trackInfoCount);
+
+
+    mStreamInfo = *pStreamInfo;
+
+    // Test - we don't have real data
+    mStreamInfo.streamCaps.nalAdaptationFlags = NAL_ADAPTATION_FLAG_NONE;
+    mStreamInfo.streamCaps.fragmentDuration = 1 * HUNDREDS_OF_NANOS_IN_A_SECOND;
+    mStreamInfo.streamCaps.storePressurePolicy = CONTENT_STORE_PRESSURE_POLICY_OOM;
+    mStreamInfo.streamCaps.viewOverflowPolicy = CONTENT_VIEW_OVERFLOW_POLICY_DROP_TAIL_VIEW_ITEM;
+    mStreamInfo.streamCaps.keyFrameFragmentation = FALSE;
+
+
+    createDefaultProducerClient(FALSE, FUNCTIONALITY_TEST_CREATE_STREAM_TIMEOUT);
+
+    EXPECT_EQ(STATUS_SUCCESS, createTestStream(0, STREAMING_TYPE_REALTIME, TEST_MAX_STREAM_LATENCY, TEST_STREAM_BUFFER_DURATION));
+    streamHandle = mStreams[0];
+
+    for (i = 0; i < totalFrames; i++) {
+        startTime = GETTIME();
+        // i = 480 will be start of a key frame, if I don't start on key-frame
+        // this test fails
+        if( (i < 30) || (i >= 480) ) {
+            mFrame.trackId = 1;
```

Comment:
You can use `DEFAULT_VIDEO_TRACK_ID` and `DEFAULT_AUDIO_TRACK_ID` instead of 1 and 2.

Reply:
Good idea actually created test `#define`'s for this.